arithmetization/src/main/java/net/consensys/linea/blockcapture/snapshots/ConflationSnapshot.java

```diff
@@ -29,4 +34,44 @@ public record ConflationSnapshot(
     List<BlockSnapshot> blocks,
     List<AccountSnapshot> accounts,
     List<StorageSnapshot> storage,
-    List<BlockHashSnapshot> blockHashes) {}
+    List<BlockHashSnapshot> blockHashes) {
+
+  /**
+   * Construct a block hash map for any block hashes embedded in this conflation.
+   *
+   * @return
+   */
+  public BlockHashLookup toBlockHashLookup() {
+    BlockHashMap map = new BlockHashMap();
+    // Initialise block hashes.  This can be null for replays which pre-date support for block hash
```

Comment:
It may be worth re-capturing them. We have lists and @gusiri 's `bulk_capture.sh` script.